volttron/platform/agent/base_historian.py

```diff
@@ -457,6 +487,9 @@ def _configure(self, config_name, action, contents):
         self._retry_period = retry_period
         self._submit_size_limit = submit_size_limit
         self._max_time_publishing = timedelta(seconds=max_time_publishing)
+        self._history_limit_days = timedelta(days=history_limit_days) if history_limit_days else None
```

Comment:
line 463 already checks "if history_limit_days"

Reply:
Yes, but that is a check for None before doing a float conversion. If it's None it stays None. Before setting the instance variable we need to check again before we create the timedelta, otherwise it will  raise a TypeError if we try to pass None as the "days" argument.